api/gateway.proto

```diff
@@ -80,7 +80,17 @@ message GatewayVersion {
   uint32 clock_source = 7;
 }
 
+// Authentication code for claiming gateways.
+message GatewayClaimAuthenticationCode {
+  option (gogoproto.populate) = false;
+
+  bytes value = 1[(validate.rules).bytes = {max_len: 2048}];
```

Comment:
This `value` should be of type Secret instead of wrapping the entire marshaled message in a Secret